src/citrine/informatics/predictor_evaluation_result.py

```diff
@@ -0,0 +1,286 @@
+
+
+class MetricValue(PolymorphicSerializable["MetricValue"]):
+    """[ALPHA] Value associated with a metric computed during a Predictor Evaluation Workflow."""
+
+    @classmethod
+    def get_type(cls, data) -> Type[Serializable]:
+        """Return the subtype."""
+        return {
+            "RealMetricValue": RealMetricValue,
+            "PredictedVsActualValue": PredictedVsActualValue,
+            "RealPredictedVsActual": RealPredictedVsActual,
+            "CategoricalPredictedVsActual": CategoricalPredictedVsActual
+        }[data["type"]]
+
+
+class RealMetricValue(Serializable["RealMetricValue"], MetricValue):
+    """[ALPHA] Mean and standard error computed for a real-valued metric.
+
+    Parameters
+    ----------
+    mean: float
+        Mean value
+    standard_error: float
+        Standard error of the mean
+
+    """
+
+    mean = properties.Float("mean")
+    standard_error = properties.Optional(properties.Float(), "standard_error")
+    typ = properties.String('type', default='RealMetricValue', deserializable=False)
+
+    def __init__(self, *,
+                 mean: float,
+                 standard_error: Optional[float]):
+        self.mean = mean
+        self.standard_error = standard_error
+
+    def __eq__(self, other):
+        if isinstance(other, RealMetricValue):
+            return self.mean == other.mean and self.standard_error == other.standard_error
+        else:
+            return False
+
+
+class PredictedVsActual(MetricValue):
+    """[ALPHA] Predicted vs. actual data computed for a single data point."""
+
+    @classmethod
+    def get_type(cls, data) -> Type[Serializable]:
+        """Return the subtype."""
+        return {
+            "RealPredictedVsActual": RealPredictedVsActual,
+            "CategoricalPredictedVsActual": CategoricalPredictedVsActual
+        }[data["type"]]
+
+
+class RealPredictedVsActual(Serializable["RealPredictedVsActual"], PredictedVsActual):
+    """[ALPHA] Predicted vs. actual data for a single real-valued data point.
+
+    Parameters
+    ----------
+    uuid: UUID
+        Unique Citrine id given to the candidate
+    identifiers: Set[str]
+        Set of globally unique identfiers given to the candidate
+    trial: int
+        1-based index of the trial this candidate belonged to
+    fold: int
+        1-based index of the fold this candidate belonged to
+    predicted: RealMetricValue
+        Predicted value
+    actual: RealMetricValue
+        Actual value
+
+    """
+
+    uuid = properties.UUID("uuid")
+    identifiers = properties.Set(properties.String, "identifiers")
+    trial = properties.Integer("trial")
+    fold = properties.Integer("fold")
+    predicted = properties.Object(RealMetricValue, "predicted")
+    actual = properties.Object(RealMetricValue, "actual")
+    typ = properties.String('type', default='RealPredictedVsActual', deserializable=False)
+
+    def __init__(self, *,
```

Comment:
Is the keyword-only arguments pattern something we're going to be applying to all of citrine?

Reply:
I'd really like to.  I tend to start with it because its easier to remove than to add.